### PR TITLE
Validate ServerOptions.MaxConnections and MaxPendingConnections

### LINKED ISSUE #4603: [Audit-Low] negative ServerOptions.MaxConnections silently disables the connection cap
> **AI-generated audit finding** — this issue was opened from an automated security/correctness audit. It **has not been triaged by a human yet**; verify the reasoning, reproducibility, and severity before acting on it.

### Low: negative \`ServerOptions.MaxConnections\` silently disables the connection cap

Affected code:

- [src/IceRpc/ServerOptions.cs:23-26](https://github.com/icerpc/icerpc-csharp/blob/main/src/IceRpc/ServerOptions.cs#L23) — \`MaxConnections\` documents \`0\` as \"unlimited\" but is an auto-property with no validation, so any \`int\` (including negative values) is accepted
- [src/IceRpc/ServerOptions.cs:33](https://github.com/icerpc/icerpc-csharp/blob/main/src/IceRpc/ServerOptions.cs#L33) — \`MaxPendingConnections\` has the same shape (no setter validation)
- [src/IceRpc/Server.cs:447](https://github.com/icerpc/icerpc-csharp/blob/main/src/IceRpc/Server.cs#L447) — the cap is only enforced when \`_maxConnections > 0\`
- [src/IceRpc/Server.cs:358-360](https://github.com/icerpc/icerpc-csharp/blob/main/src/IceRpc/Server.cs#L358) — \`MaxPendingConnections\` flows into \`new SemaphoreSlim(_maxPendingConnections, _maxPendingConnections)\`; a negative value throws \`ArgumentOutOfRangeException\` from \`SemaphoreSlim\` rather than failing at option validation time

### Verification

Confirmed. \`ServerOptions.MaxConnections = -1\` is accepted. \`Server\` only blocks new connections when \`_maxConnections > 0 && _connections.Count + _detachedConnectionCount == _maxConnections\`, so a negative value behaves identically to the documented \"unlimited\" value of \`0\`. The same gap exists for \`MaxPendingConnections\`, where a negative value fails later inside \`SemaphoreSlim\` construction with an obscure exception instead of a clear validation error.

### Impact

- An operator who intended to configure a finite connection cap can accidentally disable it through configuration binding, a typo, or an off-by-one calculation, weakening protection against connection-exhaustion attacks.
- For \`MaxPendingConnections\`, a misconfigured negative value crashes the listen loop with a \`SemaphoreSlim\` argument exception that does not point at the configuration error.

### Recommendation

- Validate \`MaxConnections >= 0\` in the property setter (or during server construction) and throw a clear \`ArgumentException\` on negative values.
- Validate \`MaxPendingConnections > 0\` in the property setter with a clear message, so misconfiguration is surfaced where the operator can fix it.
- Add tests that constructing a \`Server\` with a negative \`MaxConnections\` / \`MaxPendingConnections\` fails fast with a useful message.

**Severity (auditor-assigned):** Low

### PR DESCRIPTION
**Pull request by pepone**
Fixes #4603.

### CODE REVIEW

File: src/IceRpc/ServerOptions.cs [hunk 1]
```diff
@@ -23,14 +23,30 @@ public TimeSpan ConnectTimeout
...
+                nameof(value),
+                value,
+                $"{nameof(MaxConnections)} must be greater than or equal to 0 (where 0 means unlimited).");
+    }
 
     /// <summary>Gets or sets the maximum number of server connections waiting for connection establishment to complete.
     /// Once the maximum number of pending connections has been reached, the server will stop accepting new connections
     /// to ensure that the transport rejects new connections once its connection backlog is full.</summary>
     /// <value>The maximum number of connection waiting for connection establishment to complete. Defaults to
     /// <c>100</c>.</value>
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply src/IceRpc/ServerOptions.cs:40]: I don't think this is necessary.

**Review by bernardnormier — APPROVED** [review]

**Review by externl — APPROVED** [review]
